x-pack/filebeat/input/httpjson/request.go

```diff
@@ -300,6 +302,24 @@ func (r *requester) doRequest(stdCtx context.Context, trCtx *transformContext, p
 			if err != nil {
 				return fmt.Errorf("failed to execute rf.collectResponse: %w", err)
 			}
+			// store first response in transform context
+			bodyMap := make(mapstr.M)
```

Comment:
Initially it was **var bodyMap mapstr.M** but changed it to **bodyMap := make(mapstr.M)** because there might be a scenario where the Unmarshall fails due to a malformed response. Now we only a log error here and we do not return the error : reason - the user use case might not depend on first_response and we don't want it to fail. In this scenario bodyMap will be nil and if accessed anywhere down the flow ... it will throw a panic.

Reply:
There are checks preventing this in most places where bodyMap is checked for a nil value ... but due to the complexity of the code thought this might be safer.

Reply:
But I see the reasoning here, It's better to fail deterministically rather than going unnoticed and create silent failures elsewhere.

Reply:
Yeah, the only way that `bodyMap` will end up being nil is if the unmarshalling fails https://play.golang.com/p/JHhe6mxuaLT and in that situation we drop this anyway.